paddle/fluid/operators/sum_op.cc

```diff
@@ -115,7 +115,7 @@ class SumOpMaker : public framework::OpProtoAndCheckerMaker {
   void Make() override {
     AddInput("X", "(vector<Tensor>) The input tensors of sum operator.")
         .AsDuplicable();
-    AddOutput("Out", "(Tensor) The output tensor of sum operator.");
+    AddOutput("Out", "(Tensor) The output tensor of sum operator.").Reuse("X");
```

Comment:
In this case, Out should reuse the first Tensor memory of X(vector<Tensor> ), but we didn't have the sub-variable name of X.